### ISSUE TITLE
Ground truth is not updated between self-play episodes

### ISSUE THREAD

**Issue by ymyluckygithub** [dmar-bonn/ipp-rl#5]
## Description

`EpisodeGenerator.__init__()` appears to regenerate the sensor simulation ground-truth map for every self-play episode:
```python
self.mapping.sensor.sensor_simulation.create_ground_truth_map()

However, the return value of create_ground_truth_map() is discarded.
The implementations of create_ground_truth_map(), such as GaussianRandomField.create_ground_truth_map(), return a newly generated NumPy array but do not update self.ground_truth_map internally:

def create_ground_truth_map(self) -> np.array:
    return ground_truths.gaussian_random_field(
        lambda k: k ** (-self.cluster_radius),
        self.sensor.grid_map.y_dim,
        self.sensor.grid_map.x_dim,
    )

As a result, the following call has no effect:

self.mapping.sensor.sensor_simulation.create_ground_truth_map()

The ground_truth_map initially created in the simulation constructor remains unchanged.
Current behavior
Within one MCTS-Zero training run, all self-play episodes and self-play iterations use the same initial ground-truth map.
The current behavior is effectively:
Create initial ground truth G0
    ├── Self-play iteration 0
    │   ├── Episode 1 uses G0
    │   ├── Episode 2 uses G0
    │   └── ...
    ├── Self-play iteration 1
    │   ├── Episode 1 uses G0
    │   └── ...
    └── Online deployment/evaluation also uses G0
Although create_ground_truth_map() is called for every episode, the newly generated map is immediately discarded.
Expected behavior
If the intention is to train on randomized environments, each self-play episode should assign the newly generated map to the simulation:
simulation = self.mapping.sensor.sensor_simulation
simulation.ground_truth_map = simulation.create_ground_truth_map()
The expected behavior would then be:
Episode 1 uses G1
Episode 2 uses G2
Episode 3 uses G3
...
Why this matters
Using the same ground-truth map throughout training may reduce environment diversity and lead to overfitting.
This is particularly relevant in adaptive informative path planning because the ground truth affects:
1. simulated sensor measurements;
2. the estimated map mean;
3. the adaptive region-of-interest mask;
4. the reward calculation;
5. the generated MCTS policy and value targets.
For non-adaptive planning, the effect may be smaller because covariance prediction does not directly depend on measurement values. However, observation trajectories and map reconstruction evaluation still use the unchanged ground truth.
It also means that when training and online planning occur during the same execute() call, the model is evaluated on the same ground-truth environment used for self-play data generation.
Location
File:
planning/mcts_zero/episode_generators.py
Current code:
self.mapping = Mapping(
    grid_map,
    sensor,
    shuffle_prior_cov=self.hyper_params["shuffle_prior_cov"],
)
self.mapping.sensor.sensor_simulation.create_ground_truth_map()
Suggested fix
Assign the generated map explicitly:
self.mapping = Mapping(
    grid_map,
    sensor,
    shuffle_prior_cov=self.hyper_params["shuffle_prior_cov"],
)

simulation = self.mapping.sensor.sensor_simulation
simulation.ground_truth_map = simulation.create_ground_truth_map()
Alternatively, introduce an explicit method that regenerates and stores the map:
class Simulation:
    def reset_ground_truth_map(self):
        self.ground_truth_map = self.create_ground_truth_map()
        return self.ground_truth_map
Then use:
self.mapping.sensor.sensor_simulation.reset_ground_truth_map()
The second option makes the intended side effect clearer and avoids callers accidentally discarding the returned map.
Additional consideration
It may be useful to separate training and evaluation environments explicitly:
- regenerate a ground truth for every self-play episode;
- use independent ground truths for validation;
- regenerate another unseen ground truth before online deployment/evaluation.
This would prevent the model from being trained and evaluated on the same environment.
Suggested test
A unit test could mock create_ground_truth_map() with a known array and verify that constructing a new EpisodeGenerator updates sensor_simulation.ground_truth_map.
For example:

new_ground_truth = np.ones_like(simulation.ground_truth_map)
simulation.create_ground_truth_map = Mock(return_value=new_ground_truth)

# Construct EpisodeGenerator or call the reset method.

np.testing.assert_array_equal(
    simulation.ground_truth_map,
    new_ground_truth,
)
